sleepyq/__init__.py

```diff
@@ -36,66 +36,208 @@ def __init__(self, data):
+            # Raise error
+            r.raise_for_status()
+        return r
 
     def login(self):
         if '_k' in self._session.params:
             del self._session.params['_k']
-
         data = {'login': self._login, 'password': self._password}
-        r = self._session.put('https://api.sleepiq.sleepnumber.com/rest/login', json=data)
-        r.raise_for_status()
-
+        r = self._session.put(self._api+'/login', json=data)
+        if r.status_code == 401:
+            return False
         self._session.params['_k'] = r.json()['key']
-
         return True
 
     def sleepers(self):
-        url = 'https://api.sleepiq.sleepnumber.com/rest/sleeper'
-        r = self._session.get(url)
-        r.raise_for_status()
-
+        url = '/sleeper'
+        r=self.__makeRequest(url)
         sleepers = [Sleeper(sleeper) for sleeper in r.json()['sleepers']]
         return sleepers
 
     def beds(self):
-        url = 'https://api.sleepiq.sleepnumber.com/rest/bed'
-        r = self._session.get(url)
-        r.raise_for_status()
+        url = '/bed'
+        r=self.__makeRequest(url)
         beds = [Bed(bed) for bed in r.json()['beds']]
         return beds
 
     def beds_with_sleeper_status(self):
         beds = self.beds()
         sleepers = self.sleepers()
         family_statuses = self.bed_family_status()
-
         sleepers_by_id = {sleeper.sleeper_id: sleeper for sleeper in sleepers}
         bed_family_statuses_by_bed_id = {family_status.bed_id: family_status for family_status in family_statuses}
-
         for bed in beds:
             family_status = bed_family_statuses_by_bed_id[bed.bed_id]
-
             for side in ['left', 'right']:
                 sleeper_key = 'sleeper_' + side + '_id'
                 sleeper_id = getattr(bed, sleeper_key)
+                if sleeper_id == "0":
+                    continue
                 sleeper = sleepers_by_id[sleeper_id]
-
                 status = getattr(family_status, side)
                 status.sleeper = sleeper
-
                 setattr(bed, side, status)
         return beds
 
 
     def bed_family_status(self):
-        url = 'https://api.sleepiq.sleepnumber.com/rest/bed/familyStatus'
-        r = self._session.get(url)
-        r.raise_for_status()
-
+        url = '/bed/familyStatus'
+        r=self.__makeRequest(url)
         statuses = [FamilyStatus(status) for status in r.json()['beds']]
         return statuses
+
+    def set_light(self, bedId, light, setting):
+        #
+        # light 1-4
+        ### 1=Right Night Stand
+        ### 2=Left Night Stand
+        ### 3=Right Night Light
+        ### 4=Left Night Light
+        # setting 0=off, 1=on
+        #
+        if 1 <= light <= 4 and setting in (0, 1):
+            url = '/bed/'+bedId+'/foundation/outlet'
+            data = {'outletId': light, 'setting': setting}
+            r=self.__makeRequest(url, "put", data)
+            return True
+        else:
+            return False
```

Comment:
If you try to tell it use something not supported at all, is it better to return false, or throw an error? Either could work as long as it's documented.

Reply:
I changed all of the "return False" to "raise ValueError()" describing the reason for the error